src/pip/_internal/utils/entrypoints.py

```diff
@@ -25,3 +41,34 @@ def _wrapper(args: Optional[List[str]] = None) -> int:
         "running pip directly.\n"
     )
     return main(args)
+
+
+def get_best_invocation_for_this_pip() -> str:
+    """Try to figure out the best way to invoke pip in the current environment."""
+    binary_directory = "Scripts" if WINDOWS else "bin"
+    binary_prefix = os.path.join(sys.prefix, binary_directory)
+
+    # Try to use pip[X[.Y]] names, if those executables for this environment are
+    # the first on PATH with that name.
+    exe_are_in_PATH = binary_prefix in os.environ.get("PATH", "").split(os.pathsep)
```

Comment:
This ought to be a case insensitive check on case insensitive systems. Note that this doesn't mean just "Windows", so it's really difficult to get this right - so as long as we're OK with being wrong in some edge cases I'm OK with being case insensitive on Windows only. But we really need to be case insenstitive on Windows, path entries seem to often end up with inconsistent case for drive letters, for instance.

Reply:
Done, with a normcase. :)